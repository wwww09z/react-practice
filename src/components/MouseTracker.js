import React, { useState } from 'react';

const MouseTracker = (props) => {
  const [mousePosition, setMousePosition] = useState({ x: 0, y: 0 });
  const handleMouseMove = (event) => {
    setMousePosition({ x: event.clientX, y: event.clientY });
  };
  return (
    <div onMouseMove={handleMouseMove} style={{ height: '100vh' }}>
      {props.render(mousePosition)}
    </div>
  );
};

export default MouseTracker;
