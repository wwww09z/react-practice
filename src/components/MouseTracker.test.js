import React from "react";
import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import MouseTracker from "./MouseTracker";

describe('MouseTracker', () => {
  it('track mouse move', () => {
    const renderMock = jest.fn(({ x, y }) => <div>Mouse position is {x}, {y}</div>);

    const { container } = render(<MouseTracker render={renderMock} />);
    fireEvent.mouseMove(container.firstChild, { clientX: 100, clientY: 200 });

    expect(renderMock).toHaveBeenCalledWith({ x: 100, y: 200 });
    expect(container.textContent).toBe('Mouse position is 100, 200');
  });

  test('check snapshot', () => {
    const renderMock = jest.fn(() => <div>Mouse position is 100, 200</div>);
    const tree = renderer.create(<MouseTracker render={renderMock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});