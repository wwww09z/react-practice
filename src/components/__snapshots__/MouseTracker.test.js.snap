// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MouseTracker check snapshot 1`] = `
<div
  onMouseMove={[Function]}
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div>
    Mouse position is 100, 200
  </div>
</div>
`;
